plugins/aws/aws.plugin.zsh

```diff
@@ -82,12 +83,14 @@ function acp() {
     echo "Assuming role $role_arn using profile $profile"
     local assume_cmd=(aws sts assume-role "--profile=$profile" "--role-arn $role_arn" "--role-session-name $profile" "$mfa_opt" "$extid_opt"
```

Comment:
I think the issue is that separate arguments need to be separated, especially `$mfa_opt` and `$extid_opt` which need to be word-split with `${=var}`:
```suggestion
    local assume_cmd=(aws sts assume-role --profile="$profile" --role-arn "$role_arn" --role-session-name "$profile" ${=mfa_opt} ${=extid_opt}
```
Also, older zsh versions won't support having the array being defined and assigned at the same time, so it should be first declared with `local -a assume_cmd`, then assigned in the next line.

Reply:
I refactored the whole function and cleaned up the logic a bit in https://github.com/ohmyzsh/ohmyzsh/pull/9426. I think this is no longer needed.

Reply:
Thanks, @mcornella! Yes, the older version of Zsh may be the reason - I vaguely remember having another issue due to Macs by default having an old version of Zsh.

I'll withdraw this PR from @robvadai, as I like yours more. 👍